vpr/src/route/rr_graph_clock.h

```diff
@@ -0,0 +1,119 @@
+#ifndef RR_GRAPH_CLOCK_H
+#define RR_GRAPH_CLOCK_H
+
+#include <string>
+#include <vector>
+#include <memory>
+#include <unordered_map>
+#include <set>
+#include <utility>
+
+#include "clock_fwd.h"
+
+#include "clock_network_types.h"
+#include "clock_connection_types.h"
+
+class ClockNetwork;
+class ClockConnection;
+
+class SwitchPoint {
+    public:
+        // [grid_width][grid_height][nodes]
+        std::vector<std::vector<std::vector<int>>> rr_node_indices;
+        std::set<std::pair<int, int>> locations; // x,y
+    public:
+        /** Getters **/
+        std::vector<int> get_rr_node_indices_at_location(int x, int y) const;
+
+        std::set<std::pair<int, int>> get_switch_locations() const;
+
+        /** Setters **/
+        void insert_node_idx(int x, int y, int node_idx);
+};
+
+class SwitchPoints {
+    public:
+        std::unordered_map<std::string, SwitchPoint> switch_name_to_switch_location;
+    public:
+        /** Getters **/
+        std::vector<int> get_rr_node_indices_at_location(
+            std::string switch_name,
```

Comment:
switch name is not a very clear name -- switches are normally edges, not nodes.  
Comment what this routine does -- exactly how to set the inputs and give an example of what is returned for a certain call (e.g. x,y = middle of chip, switch_name == name of main drive of global clock spine, returns rr-nodes of all the clock spines that start networks there).

Reply:
Done: commit ae598aecdccc87b08a7fd15511fbd39e4651a7c2